src/mlpack/tests/decision_tree_test.cpp

```diff
@@ -701,6 +701,66 @@ TEST_CASE("SimpleGeneralizationTest", "[DecisionTreeTest]")
   REQUIRE(wdcorrect > 0.75);
 }
 
+/**
+ * Test that the decision tree generalizes reasonably when built on float data.
+ */
+TEST_CASE("SimpleGeneralizationFMatTest", "[DecisionTreeTest]")
+{
+  arma::fmat inputData;
+  if (!data::Load("vc2.csv", inputData))
+    FAIL("Cannot load test dataset vc2.csv!");
+
+  arma::Row<size_t> labels;
+  if (!data::Load("vc2_labels.txt", labels))
+    FAIL("Cannot load labels for vc2_labels.txt");
+
+  // Initialize an all-ones weight matrix.
+  arma::rowvec weights(labels.n_cols, arma::fill::ones);
+
+  // Build decision tree.
+  DecisionTree<> d(inputData, labels, 3, 10); // Leaf size of 10.
+  DecisionTree<> wd(inputData, labels, 3, weights, 10); // Leaf size of 10.
```

Comment:
```suggestion
  DecisionTree<> d(inputData, labels, 3, 10 /* Leaf size of 10. */);
  DecisionTree<> wd(inputData, labels, 3, weights, 10 /* Leaf size of 10. */);
```

Wondering if attaching the comment to the parameter makes it more clear?

Reply:
Ah, nice, agreed. :+1: